lib/pure/typetraits.nim

```diff
@@ -71,46 +69,6 @@ proc distinctBase*(T: typedesc): typedesc {.magic: "TypeTrait".}
   ## Returns base type for distinct types, works only for distinct types.
   ## compile time error otherwise
 
-import std/macros
-
-macro lenTuple*(t: tuple): int {.since: (1, 1).} =
-  ## Return number of elements of `t`
-  newLit t.len
-
-macro lenTuple*(t: typedesc[tuple]): int {.since: (1, 1).} =
-  ## Return number of elements of `T`
-  newLit t.len
-
-since (1, 1):
-  template get*(T: typedesc[tuple], i: static int): untyped =
-    ## Return `i`th element of `T`
-    # Note: `[]` currently gives: `Error: no generic parameters allowed for ...`
-    type(default(T)[i])
-
-macro genericParams*(T: typedesc): untyped {.since: (1, 1).} =
```

Comment:
This shouldn't be removed IMO. The other stuff I agree should be removed

Reply:
I think it should be removed. The informal reason is: ``shitty typedesc``. There is even a PR that I am currently working on that would make this solution impossible https://github.com/nim-lang/Nim/pull/11959

Reply:
Can you point out whats bad about it here? (sorry to disrupt your crusade :p)
I think preventing to return typedesc is a very bad idea. I can understand it for procs, but not for templates/macros.

Reply:
nothing against returning typedesc from a macro. But putting typedesc in any form of container, such as array or tuple is bad. What do you think ``(int,float)`` is? Is it the type for a tuple of int and float, or is it a tuple containing the type `int` and the type `float`? If you print the result of this proc you will get somthing that will be something different when parsed. There are probably more problems under the hood that I can't name, because typedesc always has hidden problems.

Reply:
I just found out that this function does return a tuple type, not a tuple value. And and since a tuple type does not have accessors, there accessors were added to `typetraits`.